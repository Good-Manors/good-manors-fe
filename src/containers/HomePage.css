.HomePageWrapper {
  width: 100vw;
  height: 100vh;
  display: grid;
  grid-template-columns: 25% 75%;
  grid-template-rows: 20% 80%;
  grid-template-areas: "header ."
                       "header drawerList";
                       
}

.DrawerListWrapper {
  display: flex;
  flex-direction: column;
  height: 90vh;
  align-content: center;
  overflow: auto;
  margin: 2em 1em;
}
