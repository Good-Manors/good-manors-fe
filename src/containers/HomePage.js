import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Header from '../components/Header/Header';
import Menu from '../components/Menu';
import Search from '../components/Search';
import DrawerList from '../components/Drawer/DrawerList';
import styles from './HomePage.css';
import { getDrawers } from '../selectors/homeSelectors';
import { getFirstHome } from '../services/homes';
import { setHome } from '../actions/homeActions';


const HomePage = () => {

  const dispatch = useDispatch();
  const drawers = useSelector(getDrawers);


  useEffect(() => {
    getFirstHome()
      .then(home => {
        dispatch(setHome(home));
      });
  }, []);



  return (
    <section id='home-page-wrapper' className={styles.HomePageWrapper}>
      <Header />
      <Menu />
      <Search />
      <section className={styles.DrawerListWrapper}>
        <DrawerList drawers={drawers || []} />
      </section>

    </section>
  );

};

export default HomePage;
