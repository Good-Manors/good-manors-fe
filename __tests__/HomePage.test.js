import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import HomePage from '../src/containers/HomePage';

describe('Renders a HomePage container', () => {
  it('Should render a HomePage', () => {
    const getWrapper = (mockStore = createStore()) => mount(
      <Provider store={mockStore}>
        <HomePage/>
      </Provider>
    );
    expect(getWrapper).toMatchSnapshot();
  });
});
