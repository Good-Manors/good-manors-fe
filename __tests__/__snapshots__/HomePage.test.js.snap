// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a HomePage container Should render a HomePage 1`] = `[Function]`;
